From 9118a4de0505712792cc3cf7a29733a6efd46f05 Mon Sep 17 00:00:00 2001
From: Balamurugan Ramar <quic_bramar@quicinc.com>
Date: Mon, 27 Nov 2023 23:54:03 +0530
Subject: [PATCH] sigma-dut: Enabling preferred AC by default

For SCS(Stream Classification Service), Preferred AC should be enabled
by default.

Preferred AC comes from the station in SCS request frame AP parses this
SCS request and configure it in firmware.

Client sends the VOICE traffic with preferred AC but preferred AC
is not applied by firmware because it is disabled in sigma-dut.

After enabling preferred AC, works fine.

Signed-off-by: Balamurugan Ramar <quic_bramar@quicinc.com>
---
 ap.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/ap.c b/ap.c
index 67e3c56..db3308e 100644
--- a/ap.c
+++ b/ap.c
@@ -8625,8 +8625,11 @@ static void fwtest_set_he_params_link_id(struct sigma_dut *dut, const char *ifna
 	fwtest_linkid_cmd_wrapper(dut, "-m 0x48 -v 0 142 1", ifname, link_id);
 	/* Enable mix bw */
 	fwtest_linkid_cmd_wrapper(dut, "-m 0x47 -v 0 141 1", ifname, link_id);
-	/* Disable preferred AC */
-	fwtest_linkid_cmd_wrapper(dut, "-m 0x48 -v 0 186 0", ifname, link_id);
+	/* Disable preferred AC for HE only */
+	if (dut->program == PROGRAM_HE)
+		fwtest_linkid_cmd_wrapper(dut, "-m 0x48 -v 0 186 0", ifname, link_id);
+	else
+		fwtest_linkid_cmd_wrapper(dut, "-m 0x48 -v 0 186 1", ifname, link_id);
 	if (dut->ap_he_ulofdma == VALUE_ENABLED) {
 		/* he_dynamic_muedca 0 */
 		run_system_wrapper(dut,
-- 
2.17.1

